crypto/rand/rand_lib.c

```diff
@@ -204,7 +206,8 @@ int RAND_set_rand_engine(ENGINE *engine)
             return 0;
         }
     }
-    CRYPTO_THREAD_write_lock(rand_engine_lock);
+    if (!CRYPTO_THREAD_write_lock(rand_engine_lock))
```

Comment:
Do we need to call ENGINE_finish() in case engine != NULL?

Reply:
`ENGINE_finish` checks for NULL, but yes.